src/backend/function/Fusion.Resources.Functions/Functions/ProfileSync.cs

```diff
@@ -0,0 +1,127 @@
+                {
+                    log.LogInformation($"Detected change in profile '{resourcesPerson.Mail}'. Initiating refresh.");
+                    var refreshResponse = await resourcesClient.PostAsJsonAsync($"resources/personnel/{resourcesPerson.Mail}/refresh", new { });
+                    refreshResponse.EnsureSuccessStatusCode();
+                }
+            }
+
+            log.LogInformation("Profiles sync run completed");
+        }
+
+        private async Task<List<PersonValidationResult>> EnsurePeople(List<string> mailsToEnsure)
+        {
+            var peopleRequest = new HttpRequestMessage(HttpMethod.Post, $"persons/ensure");
+            peopleRequest.Headers.Add("api-version", "2.0");
+            var bodyContent = JsonConvert.SerializeObject(new { PersonIdentifiers = mailsToEnsure });
+            peopleRequest.Content = new StringContent(bodyContent, Encoding.UTF8, "application/json");
+
+            var peopleResponse = await peopleClient.SendAsync(peopleRequest);
+            peopleResponse.EnsureSuccessStatusCode();
+
+            var peopleBody = await peopleResponse.Content.ReadAsStringAsync();
+            var people = JsonConvert.DeserializeObject<List<PersonValidationResult>>(peopleBody);
+            var ensuredPeople = people.Where(p => p.Success).ToList();
+            return ensuredPeople;
+        }
+
+        private bool InvitationStatusMatches(InvitationStatus? peopleStatus, ApiAccountStatus resourcesStatus)
+        {
+            if ((peopleStatus == InvitationStatus.Accepted && resourcesStatus == ApiAccountStatus.Available) ||
+                (peopleStatus == InvitationStatus.Pending && resourcesStatus == ApiAccountStatus.InviteSent) ||
+                (peopleStatus == InvitationStatus.NotSent && resourcesStatus == ApiAccountStatus.NoAccount) ||
+                (peopleStatus == null && resourcesStatus == ApiAccountStatus.NotSet))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public class PersonValidationResult
+        {
+            public bool Success { get; set; }
+            public int StatusCode { get; set; }
+            public string Message { get; set; }
+            public Person Person { get; set; }
+            public string Identifier { get; set; }
+        }
+
+        public class Person
+        {
+            public string Mail { get; set; }
+            public InvitationStatus? InvitationStatus { get; set; }
+        }
+
+        public class ApiPersonnel
+        {
+            public Guid Id { get; set; }
+            public Guid? AzureUniquePersonId { get; set; }
+            public string Mail { get; set; }
+            public ApiAccountStatus AzureAdStatus { get; set; }
+        }
+
+        public enum InvitationStatus { Accepted, Pending, NotSent }
+        public enum ApiAccountStatus { Available, InviteSent, NoAccount, NotSet }
```

Comment:
Not critical, but this is not a valid enum value :)

Reply:
Forgot about this one, will fix.